pkgs/development/python-modules/torch/default.nix

```diff
@@ -611,6 +611,23 @@ buildPythonPackage rec {
       install_name_tool -change @rpath/libc10.dylib $lib/lib/libc10.dylib $lib/lib/libshm.dylib
     '';
 
+  # See https://github.com/NixOS/nixpkgs/issues/296179
+  #
+  # This is a quick hack to add `libnvrtc` to the runpath so that torch can find
+  # it when it is needed at runtime.
+  extraRunpaths = [ "${lib.getLib cudaPackages.cuda_nvrtc}/lib" ];
```

Comment:
This needs to be wrapped in `optionals`, otherwise the vanilla nixpkgs instance's `torch` would also depend on non-free cuda

Reply:
Good point. Will fix.